ensemble/ensemble.go

```diff
@@ -0,0 +1,184 @@
+package ensemble
+
+import (
+	"context"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/network"
+	microcks "microcks.io/testcontainers-go"
+	"microcks.io/testcontainers-go/ensemble/postman"
+)
+
+const (
+	defaultNetworkAliasMicrocks = "microcks"
+	defaultNetworkAliasPostman  = "postman"
+)
+
+// Option represents an option to pass to the ensemble
+type Option func(*MicrocksContainersEnsemble) error
+
+// ContainerOptions represents the container options
+type ContainerOptions struct {
+	list []testcontainers.ContainerCustomizer
+}
+
+// Add adds an option to the list
+func (co *ContainerOptions) Add(opt testcontainers.ContainerCustomizer) {
+	co.list = append(co.list, opt)
+}
+
+// MicrocksContainersEnsemble represents the ensemble of containers
+type MicrocksContainersEnsemble struct {
+	ctx context.Context
+
+	network *testcontainers.DockerNetwork
+
+	microcksContainer        *microcks.MicrocksContainer
+	microcksContainerOptions ContainerOptions
+
+	postmanEnabled          bool
+	postmanContainer        *postman.PostmanContainer
+	postmanContainerOptions ContainerOptions
+}
+
+// GetNetwork returns the ensemble network
+func (ec *MicrocksContainersEnsemble) GetNetwork() *testcontainers.DockerNetwork {
+	return ec.network
+}
+
+// GetMicrocksContainer returns the Microcks container
+func (ec *MicrocksContainersEnsemble) GetMicrocksContainer() *microcks.MicrocksContainer {
+	return ec.microcksContainer
+}
+
+// GetPostmanContainer returns the Postman container
+func (ec *MicrocksContainersEnsemble) GetPostmanContainer() *postman.PostmanContainer {
+	return ec.postmanContainer
+}
+
+// Terminate helps to terminate all containers
+func (ec *MicrocksContainersEnsemble) Terminate(ctx context.Context) error {
+	// Main Microcks container
+	if err := ec.microcksContainer.Terminate(ctx); err != nil {
+		return err
+	}
+
+	// Postman container
+	if ec.postmanEnabled {
+		if err := ec.postmanContainer.Terminate(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// RunContainers creates instances of the Microcks and necessaries tools.
+// Using sequential start to avoid resource contention on CI systems with weaker hardware.
+func RunContainers(ctx context.Context, opts ...Option) (*MicrocksContainersEnsemble, error) {
+	var err error
+
+	ensemble := &MicrocksContainersEnsemble{ctx: ctx}
+
+	// Options
+	defaults := []Option{WithDefaultNetwork()}
+	options := append(defaults, opts...)
+	for _, opt := range options {
+		if err = opt(ensemble); err != nil {
+			return nil, err
+		}
+	}
+
+	// Microcks container
+	if ensemble.postmanEnabled {
+		ensemble.microcksContainerOptions.Add(microcks.WithEnv("POSTMAN_RUNNER_URL", "http://postman:3000"))
```

Comment:
Shouldn't you use the `defaultNetworkAliasPostman` const of the one picked by user? (if you let this possible...)